pkg/frontend/mysql_cmd_executor.go

```diff
@@ -2158,6 +2179,94 @@ func (cwft *TxnComputationWrapper) Compile(requestCtx context.Context, u interfa
 	if err != nil {
 		return nil, err
 	}
+	// check if it is necessary to initialize the temporary engine
+	if cwft.compile.NeedInitTempEngine(cwft.ses.IfNeedInitTempEngine()) {
+		// 0. init memory-non-dist engine
+		ck := clock.DefaultClock()
+		if ck == nil {
+			ck = clock.NewHLCClock(func() int64 {
+				return time.Now().Unix()
+			}, math.MaxInt)
+		}
+		shard := logservicepb.DNShardInfo{
+			ShardID:   2,
```

Comment:
I thought I reviewed this code, what is 2?   Why 2?

Reply:
These parameters are not actually used by the memory engine, but only for initialization.

Reply:
> I thought I reviewed this code, what is 2? Why 2?

these are from initMemEngineNonDist function.

Reply:
> I thought I reviewed this code, what is 2? Why 2?
see here
https://github.com/matrixorigin/matrixone/blob/ed991763eed40590a0d0a8740dbe8bb2b06494a9/pkg/cnservice/memory_engine.go#L64

Reply:
Arbitrary value is OK since it's single sharded. Change it to 0xbeef?